Tools: New tag for plugin release not created successfully on svn

### Description

There is currently an issue with creating a new release tag (e.g. `16.8.0`) on svn during the Gutenberg release process. Everything else in the workflow runs fine, and the plugin is successfully uploaded to the plugin directory, but the new tag isn't created.

I believe the issue happens during this workflow: [Update Changelog and upload Gutenberg plugin to WordPress.org plugin repo](https://github.com/WordPress/gutenberg/blob/trunk/.github/workflows/upload-release-to-plugin-repo.yml).

Here is an example changeset which has manually created the tag: https://plugins.trac.wordpress.org/changeset/2978003/

cc. @ockham who has helped out many times with a manual fix post-release 🙇

### Step-by-step reproduction instructions

I'm not sure how to _test_ this as I've only ever seen it happen in production 😅

It happens when running the "Publish release" workflow as [described here](https://developer.wordpress.org/block-editor/contributors/code/release/#publishing-the-release).

### Screenshots, screen recording, code snippet

_No response_

### Environment info

_No response_

### Please confirm that you have searched existing issues in the repo.

Yes

### Please confirm that you have tested with all plugins deactivated except Gutenberg.

Yes

Adding some more information from the [16.8.0 workflow](https://github.com/WordPress/gutenberg/actions/runs/6486992828/job/17616360916#step:7:422):

```
[Sending all the files]
Sending        lib/load.php
Sending        readme.txt
Transmitting file data ..................................................................................................................................................................................................................................................................................................................................................................................................................done
Committing transaction...
svn: E175012: Commit failed (details follow):
svn: E175012: Connection timed out
Error: Process completed with exit code 1.
```

--- a/.github/workflows/upload-release-to-plugin-repo.yml
+++ b/.github/workflows/upload-release-to-plugin-repo.yml
@@ -200,7 +200,8 @@ jobs:
                   svn st | grep '^?' | awk '{print $2}' | xargs -r svn add
                   svn st | grep '^!' | awk '{print $2}' | xargs -r svn rm
                   svn commit -m "Committing version $VERSION" \
-                   --no-auth-cache --non-interactive  --username "$SVN_USERNAME" --password "$SVN_PASSWORD"
+                   --no-auth-cache --non-interactive  --username "$SVN_USERNAME" --password "$SVN_PASSWORD" \
+                   --config-option=servers:global:http-timeout=60
 
             - name: Create the SVN tag
               working-directory: ./trunk
@@ -254,4 +255,5 @@ jobs:
             - name: Add the new version directory and commit changes to the SVN repository
               run: |
                   svn import "$VERSION" "$PLUGIN_REPO_URL/tags/$VERSION" -m "Committing version $VERSION" \
-                  --no-auth-cache --non-interactive  --username "$SVN_USERNAME" --password "$SVN_PASSWORD"
+                  --no-auth-cache --non-interactive  --username "$SVN_USERNAME" --password "$SVN_PASSWORD" \
+                  --config-option=servers:global:http-timeout=60

--- a/.github/workflows/upload-release-to-plugin-repo.yml
+++ b/.github/workflows/upload-release-to-plugin-repo.yml
@@ -201,7 +201,7 @@ jobs:
                   svn st | grep '^!' | awk '{print $2}' | xargs -r svn rm
                   svn commit -m "Committing version $VERSION" \
                    --no-auth-cache --non-interactive  --username "$SVN_USERNAME" --password "$SVN_PASSWORD" \
-                   --config-option=servers:global:http-timeout=60
+                   --config-option=servers:global:http-timeout=300
 
             - name: Create the SVN tag
               working-directory: ./trunk
@@ -256,4 +256,4 @@ jobs:
               run: |
                   svn import "$VERSION" "$PLUGIN_REPO_URL/tags/$VERSION" -m "Committing version $VERSION" \
                   --no-auth-cache --non-interactive  --username "$SVN_USERNAME" --password "$SVN_PASSWORD" \
-                  --config-option=servers:global:http-timeout=60
+                  --config-option=servers:global:http-timeout=300

